packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -952,6 +960,39 @@ export abstract class PBRBaseMaterial extends PushMaterial {
         this.prePassConfiguration = new PrePassConfiguration();
     }
 
+    private async _initShaderSourceAsync(forceGLSL = false) {
+        const engine = this.getScene().getEngine();
+
+        if (engine.isWebGPU && !forceGLSL) {
+            // Switch main UBO to non UBO to connect to leftovers UBO in webgpu
+            if (this._uniformBuffer) {
+                this._uniformBuffer.dispose();
+            }
+            this._uniformBuffer = new UniformBuffer(engine, undefined, undefined, this.name, true);

```

Comment:
Can you explain this change? I don't understand how it works if `UniformBuffer` is not configured as a UBO in WebGPU.

Reply:
you did a superb work on webgpu my dear! you can be proud!